runtime/onert/core/src/exec/Execution.cc

```diff
@@ -178,5 +275,155 @@ ir::Shape Execution::getOutputShape(ir::IOIndex ind) const
   return output_desc->info.shape();
 }
 
+static NNFW_TYPE datatype_to_nnfw_dtype(onert::ir::DataType dt)
+{
+  using onert::ir::DataType;
+  switch (dt)
+  {
+    case DataType::FLOAT32:
+      return NNFW_TYPE_TENSOR_FLOAT32;
+    case DataType::INT32:
+      return NNFW_TYPE_TENSOR_INT32;
+    case DataType::QUANT_UINT8_ASYMM:
+      return NNFW_TYPE_TENSOR_QUANT8_ASYMM;
+    case DataType::BOOL8:
+      return NNFW_TYPE_TENSOR_BOOL;
+    case DataType::UINT8:
+      return NNFW_TYPE_TENSOR_UINT8;
+    case DataType::INT64:
+      return NNFW_TYPE_TENSOR_INT64;
+    case DataType::QUANT_INT8_ASYMM:
+      return NNFW_TYPE_TENSOR_QUANT8_ASYMM_SIGNED;
+    case DataType::UINT32:
+    case DataType::QUANT_INT8_SYMM:
+    default:
+      throw std::runtime_error("Error: Model has type that runtime API does not support.");
+  }
+}
+
+void Execution::asyncIoDescSemWait()
+{
+  sem_wait(&_async_io_descs_sem);
+}
+
+void Execution::asyncIoDescSemPost()
+{
+  sem_post(&_async_io_descs_sem);
+}
+
+void Execution::runInference()
+{
+  uint32_t inference_cnt = 0;
+  uint32_t output_sz = primary_subgraph().getOutputs().size();
+  while (true)
+  {
+    if (isEmptyQueue())
+    {
+      if (isFinished())
+      {
+        if (!next_exes.empty())
+        {
+          for (uint32_t i = 0; i < next_exes.size(); i++)
+          {
+            std::get<0>(next_exes[i])->setFinish();
+          }
+        }
+        else
+        {
+          sholudStop();
+        }
+        break;
+      }
+    }
+    else
+    {
+      ioDescSemWait();
+      for (uint32_t i = 0; i < output_sz; i++)
+      {
+        auto opidx = primary_subgraph().getOutputs().at(i);
+        auto shape = primary_subgraph().operands().at(opidx).shape();
+        auto dtype = datatype_to_nnfw_dtype(primary_subgraph().operands().at(opidx).typeInfo().type());
```

Comment:
I wonder that type conversion is needed.

Reply:
good suggestion : )
I agree with you
It doesn't look necessary